src/js/view/controls/components/menu/menu.js

```diff
@@ -294,51 +342,56 @@ export default class Menu extends Events {
         menu.destroy();
     }
     open(evt) {
-        if (this.visible && !this.openMenus) {
+        const mainMenuVisible = this.mainMenu.visible;
+        let focusEl;
+        if (!this.items.length || this.children) {
             return;
         }
-        backButtonTarget = null;
-        let focusEl;
         if (this.isSubmenu) {
+            const sourceEvent = evt.sourceEvent || event;
+            const firstItem = this.topbar ? this.topbar.firstChild : this.items[0].el;
+            const lastItem = this.items[this.items.length - 1].el;
+            const isKeydown = sourceEvent.type === 'keydown';
+            const key = isKeydown && sourceEvent.key.replace(/(Arrow|ape)/, '');
+            const itemTarget = isKeydown && key === 'Up' ? lastItem : firstItem;
+            if (this.visible && !this.openMenus.length) {
+                if (this.items.length && isKeydown) {
+                    itemTarget.focus();
+                }
+                return;
+            }
             const { mainMenu, parentMenu, categoryButton } = this;
-            if (parentMenu.openMenus.length) {
-                parentMenu.closeChildren();
+            parentMenu.openMenus.push(this.name);
```

Comment:
if you push this after do you still need the conditional or do you need `closeChildren` to take the newest pushed menu into account here?

Reply:
yeah seems like you could also avoid this new parameter by changing the order of these two lines

Reply:
This pattern fixes an issue. Basically, when you close all of a menu's child menus (like the settings menu) it automatically closes that menu. This is to prevent that reflow, by making sure there's always an open menu in the array when switching from one top level to another.

Reply:
oh neat, good stuff